nuxeo-features/nuxeo-automation/nuxeo-automation-test/src/test/java/org/nuxeo/ecm/automation/core/operations/services/directory/TestSuggestDirectoryEntries.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.nuxeo.ecm.automation.core.operations.services.directory;
+
+import static org.junit.Assert.assertNotNull;
+
+import java.util.Collections;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.automation.AutomationService;
+import org.nuxeo.ecm.automation.OperationContext;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.automation.test.AutomationFeature;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.core.test.DefaultRepositoryInit;
+import org.nuxeo.ecm.core.test.annotations.Granularity;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+import com.google.inject.Inject;
+
+@RunWith(FeaturesRunner.class)
+@Features({ CoreFeature.class, AutomationFeature.class })
+@RepositoryConfig(init = DefaultRepositoryInit.class, cleanup = Granularity.METHOD)
+@Deploy("org.nuxeo.ecm.core.schema")
+@Deploy("org.nuxeo.ecm.directory")
+@Deploy("org.nuxeo.ecm.directory.api")
+@Deploy("org.nuxeo.ecm.directory.multi")
+@Deploy("org.nuxeo.ecm.directory.types.contrib")
+@Deploy("org.nuxeo.ecm.automation.test.test:multi-nature-contrib.xml")
+@Deploy("org.nuxeo.ecm.automation.test.test:test-user-directories-contrib.xml")
+@Deploy("org.nuxeo.ecm.default.config:OSGI-INF/nxdirectories-contrib.xml")
+public class TestSuggestDirectoryEntries {
+
+    @Inject
+    AutomationService service;
+
+    @Inject
+    OperationContext context;
+
+    @Test
+    public void testNature() throws OperationException {
+        testSuggestDirectoryEntries("nature");
+    }
+
+    @Test
+    public void testNatureMulti() throws OperationException {
+        testSuggestDirectoryEntries("natureMulti");
+    }
+
+    private void testSuggestDirectoryEntries(String directoryName) throws OperationException {
+        Blob entries = (Blob) service.run(
+                context,
+                SuggestDirectoryEntries.ID,
+                Collections.singletonMap("directoryName", directoryName));
+        assertNotNull(entries);
```

Comment:
The test would probably be more useful if there was a more detailed assertion on what is returned exactly, not just that something is returned.

Reply:
As the return type is a JSONBlob, I'm not sure what to test since parsing JSON here would be overkill to me.
Any suggestion ?

Reply:
It all depends on what the test is testing. Parsing JSON to make sure something expected is really there is really not overkill.